lib/types.nix

```diff
@@ -727,7 +727,7 @@ rec {
     # Either value of type `t1` or `t2`.
     either = t1: t2: mkOptionType rec {
       name = "either";
-      description = "${t1.description} or ${t2.description}";
+      description = "(${t1.description}) or (${t2.description})";
```

Comment:
`either` is associative, so we don't need to parenthesize its children when chained, when the child `.name` is `"either"`.

Reply:
That is true, but in practice long eithers are very hard to parse as well (see example)